### PR TITLE
Revert backtrace ban and start printing backtraces from the runtime again

### PR DESCRIPTION
**Pull request by kubamracek**
1) Revert the backtrace ban
2) Bring back the non-related fixed from that patch.
3) Change swift_reportError to only print backtraces in assert builds of the runtime.  But warning() will still print backtraces.

### CODE REVIEW

**Comment by kubamracek** [comment]
@swift-ci please test

File: test/Runtime/crash_without_backtrace.swift [hunk 1]
```diff
@@ -1,22 +0,0 @@
-// RUN: rm -rf %t
```
Comment (gottesmm) [inline test/Runtime/crash_without_backtrace.swift]: Why is this going away?

Comment (gottesmm) [inline test/Runtime/crash_without_backtrace.swift]: (just the test). I imagine we want this test still.

**Comment by gottesmm** [comment]
Both crash tests, please leave them both in = ).

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
Hm, this failed the on Linux bot:
21:48:29 Failing Tests (2):
21:48:29     Swift(linux-x86_64) :: Runtime/linux-fatal-backtrace.swift
21:48:29     Swift(linux-x86_64) :: Runtime/backtrace.swift

**Comment by kubamracek** [comment]
Ah, the Darwin bot builds with "--assertions", but the Linux bot uses "--assertions --no-swift-stdlib-assertions".

**Comment by kubamracek** [comment]
@swift-ci please test

**Pull request merged**